### ISSUE TITLE
DateField & TimeField: Text property is fragile

### ISSUE THREAD

**Issue by tig** [migueldeicaza/gui.cs#617]
These views are quite fragile. E.g. setting `Text` on them to something other than a perfectly formed date or time causes an exception.

They should take poorly formed text, and internally note an invalid state and (perhaps) display that something is wrong (e.g. `Invalid`). 

I think they should use `DateTime` as their internal representation of state instead of the way they currently use Text.

**Comment by BDisp** [comment]
They are derived from `TextField `and so they must use the `Text `property to show the date and there is a `DateTime` `Date` property to set the date. They really have rigid format that must be respect otherwise return invalid date. I already performed an improvment on date validation neither .Net provided what I needed. What you suggest to improved or repos some issue you are dealing.

**Comment by tig** [comment]
I think trying to edit DATES using a text editor idiom is toublesome. The problem is nobody around the world can agree on a text representation. The varying format options mean it's a unsolvable problem. This is why most GUI platforms have implemented "date pickers" where a calendar is presented.

I think we should do the bare-minimum to make `DateField` work Ok, and then get someone to invest in building a killer "DatePicker".

For TIME, the problem is slightly more tractable, but I will still argue that a "TimePicker" is what users actually want. 

So, in short, let's do the bare minimum to make these current classes basically work. 

To address my specific issue, I think adding code to `SetText` that simply aborts if the Text can't be parsed would be ok for now.
